crates/editor/src/editor.rs

```diff
@@ -986,6 +980,290 @@ impl CompletionsMenu {
             .into_any_element()
     }
 
+    fn truncate_completion(
+        style: &EditorStyle,
+        cx: &mut ViewContext<Editor>,
+        mat: &StringMatch,
+        completion: &mut Completion,
+        documentation: &Option<Documentation>,
+        max_completion_len: Pixels,
+    ) -> (bool, StyledText, StyledText) {
+        let highlights = gpui::combine_highlights(
+            mat.ranges().map(|range| (range, FontWeight::BOLD.into())),
+            styled_runs_for_code_label(&completion.label, &style.syntax).map(
+                |(range, mut highlight)| {
+                    // Ignore font weight for syntax highlighting, as we'll use it
+                    // for fuzzy matches.
+                    highlight.font_weight = None;
+                    (range, highlight)
+                },
+            ),
+        );
+
+        let mut inline_documentation_exists = false;
+
+        let mut documentation_text = if let Some(Documentation::SingleLine(text)) = documentation {
+            inline_documentation_exists = true;
+            text
+        } else {
+            ""
+        }
+        .to_owned();
+        let documentation_style = style.clone().text;
+
+        let documentation_highlight_style = HighlightStyle {
+            color: Some(Color::Muted.color(cx)),
+            ..Default::default()
+        };
+
+        let documentation_highlights = vec![(
+            Range {
+                start: 0,
+                end: documentation_text.len(),
+            },
+            documentation_highlight_style,
+        )];
+        let documentation_label = StyledText::new(documentation_text.clone())
+            .with_highlights(&documentation_style, documentation_highlights);
+
+        let mut completion_label =
+            StyledText::new(completion.label.text.clone()).with_highlights(&style.text, highlights);
+
+        let font_size = style.text.font_size.to_pixels(cx.rem_size());
+
+        let mut variable_name_end = completion.label.filter_range.end;
+        let mut completion_label_text = completion.label.text.clone();
+        let mut variable_name_length_truncated: i32 = 0;
+        if let Ok(ellipsis_width) =
+            cx.text_system()
+                .layout_line("...", font_size, &[style.text.to_run("...".len())])
```

Comment:
```suggestion
                .layout_line("…", font_size, &[style.text.to_run("…".len())])
```

Let's use the ellipsis character here (most users use monospace fonts, so that should be 1/3 as wide).